### PR TITLE
[New] Add pwnjs@1.0.0 with git auto-update

### LINKED ISSUE #12184: [Request] Add pwnjs
**Library name:** pwnjs
**Library description:** A Javascript library for browser exploitation
**Git repository url:** https://github.com/theori-io/pwnjs
**License (List them all if it's multiple):** MIT

---
### ℹ️  introduction
CDNJS is one of the most famous free and public web front-end CDN services. We'd like to host a new web front-end library on CDNJS so that web developer can easily use this library on our free CDN without any additional download/upload process.

### 📚  How to add a library
There are two ways to send a pull request(PR):
- [ ] 1. [**Adding a new library via a single package.json**](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#e-adding-a-new-library-by-a-single-packagejson)
  1. Creating a `package.json` under `ajax/libs/<LIBRARY_NAME>`
  2. Edit the `package.json`.
      - Required fields: `name`, `description`, `filename`, `license`, `repository`, `author`, `keywords`, auto-update config
      - Non-Required fields (If there is no field of the upstream): `homepage`
  3. Save the file and send a PR with the commit message as the followings example with npm auto-update (if you use git auto-update, please change from npm to git):
  This will be useful to us to maintain the git log and trace the history.
```
Add <LIBRARY_NAME> w/ npm auto-update via single package.json

close <#THIS_ISSUE_NUMBER>, cc <@AUTHOR>
```

- [ ] 2. [**Adding a new library with its assets**](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#d-adding-a-new-library-with-its-assets)
  1. Creating a <LIBRARY_NAME> folder under `ajax/libs/`
  2. Edit the `package.json` under `ajax/libs/<LIBRARY_NAME>`.
      - Required fields: `name`, `description`, `filename`, `version`, `license`, `repository`, `author`, `keywords`, auto-update config
      - Non-Required fields (If there is no field of the upstream): `homepage`
  3. Create a folder named after the latest release version under `ajax/libs/<LIBRARY_NAME>`.
  4. Add the files from upstream to `ajax/libs/<LIBRARY_NAME>/<VERSION>`. The file with `md5sum` should as same as upstream.
  5. Save the file and send a PR with the commit message as the followings example with npm auto-update (if you use git auto-update, please change from npm to git):
  This will be useful to us to maintain the git log and trace the history.
```
Add <LIBRARY_NAME>@<VERSION> w/ npm auto-update

close <#THIS_ISSUE_NUMBER>, cc <@AUTHOR>
```

If you still don't know how to add a library, here is a tutorial in [document](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md), hope this will help you to understand.

### 💻  How to send a pull request
There are two ways to add a library, via GitHub website or locally.
- [ ] 1. **GitHub website**
- [ ] 2. **locally (Use Git command)** 
  - Using **sparseCheckout**
    Because cdnjs's repo is too huge, currently. It's better to use `sparseCheckout` to lets you only check out the files you want so that the whole directory can only occupy 2.2 GB in your computer. Here is a [tutorial](https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md) which let you know how to enable sparseCheckout.
  - If you don't know how to create a pull request locally, please take a look at this [tutotial](https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github).

### 🤖  How to setup auto-update config
The auto-update config can automatically add libraries to the latest version easily. [Here](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md) is the document which can help you to know how to setup auto-update config. 
BTW, in `files` field, we'd like to let it be flexible but if there are some files not necessary to host, we will use some pattern to match and exclude them. There are many example configurations in [document](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md), hope this will help you to understand. 

### :memo: Note
If you are interested in joinning us and help add this library, please feel free to check what way you'd like to **add a library** and  **send a pull request**. And let me know what kind of help you'll need, we'll help you send the valid pull request, thank you. 😃

### PR DESCRIPTION
**Pull request by Billy4195**
close #12184, cc @brianairb

Pull request for issue: #12184 
Related issue(s): 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/theori-io/pwnjs
 * Official website (optional, not the repository): http://theori.io/pwnjs/
 * NPM package url (optional):
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 14
   - Count of stars: 325
   - Count of forks: 54
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/pwnjs/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "pwnjs",
+  "filename": "dist/pwn.min.js",
```
Comment (extend1994) [inline ajax/libs/pwnjs/package.json]: Just use `pwn.min.js` here. You can regard `basePath` in auto-update config as a root.

**Comment by Billy4195** [comment]
@extend1994 I has fixed this problem.

File: ajax/libs/pwnjs/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  "filename": "pwn.min.js",
+  "description": "A Javascript library for browser exploitation",
+  "version": "1.0.0",
+  "keywords": [
+    "pwn",
+    "exploitation"
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "https://github.com/theori-io/pwnjs.git",
```
Comment (sufuf3) [inline ajax/libs/pwnjs/package.json]: I prefer to using git protocol to get the files.
I'll help update.

File: ajax/libs/pwnjs/package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        "files": [
+          "**/*"
+        ]
+      }
+    ]
+  },
+  "license": "MIT",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/theori-io/pwnjs"
```
Comment (sufuf3) [inline ajax/libs/pwnjs/package.json]: I'll help let the url as the same with upstream https://github.com/theori-io/pwnjs/blob/master/package.json#L29

**Comment by PeterDaveHello** [comment]
@sufuf3 isn't http://theori.io/pwnjs/  more the document than homepage?

**Comment by sufuf3** [comment]
@PeterDaveHello I found https://github.com/theori-io/pwnjs/blob/master/package.json#L36 the README is the same with http://theori.io/pwnjs/ content, so I put it as the homepage.
Can't it as the homepage?
Thanks

**Comment by PeterDaveHello** [comment]
It's more like documentation as I say, the only reason that I can accept it it's just because of its `package.json` mentions that.

**Comment by Billy4195** [comment]
@PeterDaveHello I removed the `homepage` info in the `package.json`, thanks.

**Comment by PeterDaveHello** [comment]
@Billy4195 thanks!

**Pull request merged**